misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt

```diff
@@ -0,0 +1,29 @@
+package misk.concurrent
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+
+/**
+ * Inject this rather than using the [Executors] factory class to create thread pools. Executors
+ * created with this factory will automatically be shut down when the service or test completes.
+ *
+ * For all functions, `nameFormat` a string in the format specified by
+ * [ThreadFactoryBuilder.setNameFormat], like "rpc-pool-%d". If the string has a single `%d`
+ * placeholder it will be assigned sequentially. Omit the placeholder if only one thread will be
```

Comment:
Should we assert the placeholder when asking for pools other than the single thread one.

Reply:
Good idea. Coming in follow-up.

Reply:
https://github.com/cashapp/misk/pull/1405